tools/test.py

```diff
@@ -1423,7 +1423,7 @@ def ProcessOptions(options):
     if options.run[0] >= options.run[1]:
       print("The test group to run (n) must be smaller than number of groups (m).")
       return False
-  if options.J:
+  if options.j == 0:
```

Comment:
This changes precedence of the `-j` and `-J` flags, e.g. if `test.py -j 3 -J` was run and `JOBS=2` then the old code would have overwritten the `j` value from `JOBS` but the new code would leave `j` unchanged. 

It's probably a rare edge case that both flags would be specified.

Reply:
If we keep it so `-J` effectively negates `-j`, we'll have to update the help text to indicate that `-J` doesn't quite have no effect at all.

If we want to preserve that behavior, then perhaps the thing to do is print a warning when someone is using both `-J` and `-j`? Or maybe print a warning every time `-J` is used?

I suspect, like you, that this is a rare edge case. I also suspect that it basically does not matter. Like, if we run with 2 processes instead of 4 (or the other way around) in a rare edge case....oh well?

@richardlau Would you be comfortable if we kept the behavior as it is in this PR but add a warning that gets printed if both `-j` and `-J` are specified?

Reply:
I'd be okay with that.

Reply:
> @richardlau Would you be comfortable if we kept the behavior as it is in this PR but add a warning that gets printed if both `-j` and `-J` are specified?

This is the behavior I've implemented in a fixup commit. PTAL.